core/cache/hash.go

```diff
@@ -92,25 +92,25 @@ func (c *hashCache) incMisses() {
 	}
 }
 
-// hash returns a hash of the yaml serialized settings.
-// If the settings are not able to be serialized an error is returned.
-func hash(settings map[string]interface{}, extra ...string) (string, error) {
-	encoded, err := yaml.Marshal(settings)
-	if err != nil {
-		return "", errors.Trace(err)
-	}
-
+// hash returns a hash of the string values.
+func hash(values ...string) (string, error) {
 	hash := sha256.New()
-	for _, s := range extra {
-		_, err = hash.Write([]byte(s))
+	for _, s := range values {
+		_, err := hash.Write([]byte(s))
 		if err != nil {
 			return "", errors.Trace(err)
 		}
 	}
-	_, err = hash.Write(encoded)
+	return hex.EncodeToString(hash.Sum(nil)), nil
+}
+
+// hashSettings returns a hash of the yaml serialized settings.
+// If the settings are not able to be serialized an error is returned.
+func hashSettings(settings map[string]interface{}, extras ...string) (string, error) {
+	encoded, err := yaml.Marshal(settings)
```

Comment:
Does yaml.Marshal sort keys/is it deterministic?

Reply:
Yes... up to a limit. That reminds me to go look at the yaml library to work out what that limit is.

Reply:
It does always seem to sort the keys. Which means I have a different bug to chase in a different area of code. But that is outside the scope of this change.